### ISSUE TITLE
Refactor: Simplify client/blocks/importer/hooks/use-site-can-migrate.ts custom hook

### ISSUE THREAD

**Issue by bogiii** [Automattic/wp-calypso#84344]
This file is kind of hard to follow... Let's refactor it and provide better naming.

https://github.com/Automattic/wp-calypso/blob/31ab100b87f326795096d2029dae10605d71c77b/client/blocks/importer/hooks/use-site-can-migrate.ts

**Comment by flootr** [comment]
Regarding my feedback here https://github.com/Automattic/wp-calypso/pull/83593#discussion_r1392310244 I think we should not be syncing state from any query hooks if possible.

**Comment by bogiii** [comment]
We switched to a different migration mechanism and are closing this ticket since it is no longer used.

cc @ouikhuan
